tests/unit/s2n_mem_usage_test.c

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+
+#include "testlib/s2n_testlib.h"
+
+#include <errno.h>
+#include <fcntl.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+#include <s2n.h>
+
+/* The number of connection pairs to allocate before measuring memory
+ * usage. The greater the value, the more accurate the end result. */
+#define CONNECTIONS 250
+
+/* This is the current memory usage per connection, we should be really
+ * careful going beyong this value. */
+#define MAX_MEM_PER_CONNECTION (136 * 1024)
+
+#define MAX_ALLOWED_MEM_DIFF ((size_t) 2 * CONNECTIONS * MAX_MEM_PER_CONNECTION)
+
+
+size_t get_vm_data_size()
```

Comment:
`size_t` is unsigned, use `ssize_t` if you want to return `-1`.

Reply:
I only need -1 as a sentinel, casting it to `size_t` will make it max `size_t` which I'm fine to use as a sentinel.

Reply:
I still think your types and sentinels should match signedness, either use `size_t` as your type with `SIZE_MAX` as your sentinel, or use `ssize_t` as your type with `-1` as your sentinel.

Reply:
or define an eror constant which is `(size_t)-1`

Reply:
Alright, changed to ssize_t